### PR TITLE
Feat/dat 1207 bcr xml hcmi update

### PR DESCRIPTION
**Pull request by madopal**

### CODE REVIEW

File: sheepdog/utils/transforms/bcr_xml_to_json.py [hunk 1]
```diff
@@ -166,38 +166,52 @@ def xpath(
...
-            result = root.xpath(path, namespaces=self.namespaces)
-        except etree.XPathEvalError:
-            result = []
-        except:
-            raise
-        rlen = len(result)
-        if rlen < 1 and expected:
+
+        # to walk, we stick them all in a list, but it'll only match the first found
+        if type(path) != list:
```
Comment (ProfOak) [inline sheepdog/utils/transforms/bcr_xml_to_json.py]: Use `isinstance()` instead of `type()`

File: sheepdog/utils/transforms/bcr_xml_to_json.py [hunk 2]
```diff
@@ -166,38 +166,52 @@ def xpath(
...
+                result = []
+            except:
+                raise
+            rlen = len(result)
+
+            # if we get a result, let's check it
+            if rlen:
+                break
+
+        if rlen < 1 and expected and type(path) != list:
```
Comment (ProfOak) [inline sheepdog/utils/transforms/bcr_xml_to_json.py]: isinstance here too

File: sheepdog/utils/transforms/bcr_xml_to_json.py [hunk 3]
```diff
@@ -344,30 +361,30 @@ def get_entity_properties(self, root, entity_type, params, entity_id=''):
...
-                path, root, single=True, text=True,
-                expected=(not self.ignore_missing_properties),
-                label='{}: {}'.format(entity_type, entity_id))
-            # optional null fields are removed
-            if result is None and prop not in\
-                    dictionary.schema[entity_type].get('required', []):
-                continue
-            props[prop] = munge_property(result, _type)
+        if 'properties' in params:
+            print dictionary.schema.keys()
```
Comment (ProfOak) [inline sheepdog/utils/transforms/bcr_xml_to_json.py]: Do these need to be in a log?

Comment (madopal) [reply sheepdog/utils/transforms/bcr_xml_to_json.py]: I'm yanking all prints, was dead code.

**Pull request merged**